### PR TITLE
Tooltip revision

### PR DESCRIPTION
**Pull request by bentidy**
fixes problems when absolute element is passed in as a trigger. 

Also fixes problem when complex markup is passed finding the correct element with the offset left, happy for feedback on this how to make it better.

@rogerfranklin2 @nwc1991

### CODE REVIEW

**Comment by johnsibly** [comment]
@alexjfno1 :+1:

File: src/components/Tooltip/Tooltip.react.js [hunk 1]
```diff
@@ -17,7 +17,9 @@ class Tooltip extends React.Component {
     };
 
     const contentStyle = {
-      backgroundColor: color
+      margin: (margin) ? margin : '15px 0 0 0',
+      backgroundColor: color,
+      pointerEvents: 'none'
```
Comment (peter-mouland) [inline src/components/Tooltip/Tooltip.react.js]: are we able to remove the inline styles and use classes instead?
especially for the bgColour and pointer events.
The margin should also be handled by classes, but by what ever code calls this 'tip' component

**Comment by bentidy** [comment]
We were getting heroku errors (http://nowtv-react-toolkit.herokuapp.com/) so have fixed it with this commit

**Comment by peter-mouland** [comment]
Forgot to mention, would it be possible to add the spread operator for props ('{...props}') to the outer tip div?

It's a thing we've done with all the other components to cut down on future PRs

**Comment by bentidy** [comment]
Its hard to see where you would put the spread as its just as likely it would be on the trigger as it would be on the actual tooltip.

No problem adding it though, where would you put it?

**Comment by peter-mouland** [comment]
I'd put it on the outermost div (https://github.com/sky-uk/nowtv-react-toolkit/pull/107/files#diff-1d251c97f725ccab431e85dc38c3d609R26). This allows people to add id's for tests and data-tracking attrs  etc 

I've also just noticed we're missing some accessibility stuff.  Could the following be added? I don't think they're big changes.
 * `role="tooltip"` on the now-tooltip div
 * `aria-describedby="tip-id"` on the now-tooltip-trigger div
   * This will mean we need an id prop as required
 * To ensure screen-readers know about the div, we would need to ensure that the HTML always exist and that we simply show/hide the div on click
   * we can do this using `aria-hidden="true"` and `aria-hidden="false"` and come cool CSS
```CSS
.now-tooltip[aria-hidden="true"] {display:none;}
.now-tooltip[aria-hidden="false"] {display:block;}
```

Adding the a11y stuff won't be breaking changes, so happy to go through this stuff in another PR tomorrow if you want?

**Comment by rogerfranklin2** [comment]
Yes another PR would be better for me as there could be a change in functionality on how we close it?

For accessibility do we need to have a close button or some hidden text so people know how to close it?

**Comment by peter-mouland** [comment]
cool, lets write an issue for accessibility 
other than that :+1:

**Pull request merged**